### PR TITLE
pointers: create `pointer2` & use it with fallback to `pointer`

### PR DESCRIPTION
**Pull request by dfakhritdinov**

### CODE REVIEW

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/SchemaConfig.scala [hunk 1]
```diff
@@ -11,6 +11,7 @@ final case class SchemaConfig(
   metadataTable: String = "metadata",
   metaJournalTable: String = "metajournal",
   pointerTable: String = "pointer",
+  pointerTable2: String = "pointer2",
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/SchemaConfig.scala:14]: `pointerTable2: String` -> `pointer2Table: String`

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements2.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import cats.data.{NonEmptyList => Nel}
+import cats.syntax.all._
+import cats.{Monad, Parallel}
+import com.evolutiongaming.kafka.journal.eventual.cassandra.CassandraHelper._
+import com.evolutiongaming.kafka.journal.eventual.cassandra.EventualCassandraConfig.ConsistencyConfig
+import com.evolutiongaming.scassandra.TableName
+import com.evolutiongaming.scassandra.syntax._
+import com.evolutiongaming.skafka.{Partition, Topic}
+
+object PointerStatements2 {
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements2.scala:12]: `Pointer2Statements`

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements2.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  object SelectTopics {
+
+    def of[F[_]: Monad: CassandraSession](
+      name: TableName,
+      consistencyConfig: ConsistencyConfig.Read
+    ): F[PointerStatements.SelectTopics[F]] = {
+
+      val query = s"""SELECT DISTINCT topic, partition FROM ${ name.toCql }""".stripMargin
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements2.scala:58]: do you need `partition` here?

**Review by t3hnar — CHANGES_REQUESTED** [review]
you also need to add `migration` - create it does not exist

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements2.scala (hunk 3 above)
Comment (dfakhritdinov) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements2.scala:58]: Cassandra does not allow distinct queries of non full primary key

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala [hunk 4]
```diff
@@ -35,6 +35,14 @@ object PointerStatements {
 
   object Insert {
 
+    def apply[F[_]: Parallel](
+      legacy: Insert[F],
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala:39]: could we abstract away from `legacy` & `other`, I'd better use `first` / `second` ?

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala [hunk 5]
```diff
@@ -182,6 +220,17 @@ object PointerStatements {
 
   object SelectTopics {
 
+    def apply[F[_]: Monad](
+      legacy: SelectTopics[F],
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala:224]: could we abstract away from `legacy` & `select `, I'd better use `first` / `second` ?

**Review by t3hnar — CHANGES_REQUESTED** [review]

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala [hunk 6]
```diff
@@ -139,6 +166,17 @@ object PointerStatements {
 
   object SelectIn {
 
+    def apply[F[_]: Monad](
+      legacy: SelectIn[F],
```
[the remark this answers is not in the document]
Comment (dfakhritdinov) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala]: done

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala (hunk 5 above)
Comment (dfakhritdinov) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements.scala]: done

**Review by t3hnar — APPROVED** [review]

**Pull request merged**